From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quentin Perret <qperret@google.com>
Date: Thu, 26 Sep 2019 12:30:35 +0100
Subject: Revert "drivers: qcom: rpmh-rsc: modularize RSC controller driver"

This reverts commit 7cbdcba442ed26b789f5b3ab18b97e3aa2686bf0.
It breaks allmodconfig:

  drivers/soc/qcom/rpmhpd.o: In function `rpmhpd_aggregate_corner':
  rpmhpd.c:(.text+0x8f8): undefined reference to `rpmh_write'
  rpmhpd.c:(.text+0x900): undefined reference to `rpmh_write_async'
  rpmhpd.c:(.text+0x9c4): undefined reference to `rpmh_write_async'
  rpmhpd.c:(.text+0xa3c): undefined reference to `rpmh_write_async'

It is not obvious that QCOM_RPMH should be tristate in the first place
when QCOM_RPMHPD needs it compiled in.

Bug: 140224784
Signed-off-by: Quentin Perret <qperret@google.com>
Change-Id: I154b2d2b369c3654c41595bad91abd5a47d15206
---
 drivers/soc/qcom/Kconfig    | 2 +-
 drivers/soc/qcom/rpmh-rsc.c | 9 +++++----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/soc/qcom/Kconfig b/drivers/soc/qcom/Kconfig
index 119a38bf373e..661e47acc354 100644
--- a/drivers/soc/qcom/Kconfig
+++ b/drivers/soc/qcom/Kconfig
@@ -102,7 +102,7 @@ config QCOM_RMTFS_MEM
 	  Say y here if you intend to boot the modem remoteproc.
 
 config QCOM_RPMH
-	tristate "Qualcomm Technologies, Inc. RPM-Hardened (RPMH) Communication driver"
+	bool "Qualcomm RPM-Hardened (RPMH) Communication"
 	depends on ARCH_QCOM && ARM64 || COMPILE_TEST
 	help
 	  Support for communication with the hardened-RPM blocks in
diff --git a/drivers/soc/qcom/rpmh-rsc.c b/drivers/soc/qcom/rpmh-rsc.c
index d7d9e1b8a4f1..e278fc11fe5c 100644
--- a/drivers/soc/qcom/rpmh-rsc.c
+++ b/drivers/soc/qcom/rpmh-rsc.c
@@ -11,7 +11,6 @@
 #include <linux/io.h>
 #include <linux/kernel.h>
 #include <linux/list.h>
-#include <linux/module.h>
 #include <linux/of.h>
 #include <linux/of_irq.h>
 #include <linux/of_platform.h>
@@ -688,7 +687,9 @@ static struct platform_driver rpmh_driver = {
 		  .of_match_table = rpmh_drv_match,
 	},
 };
-builtin_platform_driver(rpmh_driver);
 
-MODULE_LICENSE("GPL v2");
-MODULE_DESCRIPTION("Qualcomm Technologies, Inc. RPMH communication driver");
+static int __init rpmh_driver_init(void)
+{
+	return platform_driver_register(&rpmh_driver);
+}
+arch_initcall(rpmh_driver_init);
